interfaces/cython/cantera/units.pyx

```diff
@@ -21,6 +21,21 @@ cdef class Units:
     def __repr__(self):
         return f"<Units({pystr(self.units.str())}) at {id(self):0x}>"
 
+    def __str__(self):
+        return pystr(self.units.str())
+
+    def dimension(self, primary):
+        return self.units.dimension(stringify(primary))
```

Comment:
One more comment: some of these methods should probably get a docstring, in which case some points raised in #1274 could be considered (i.e. tagging new features with the Sphinx directive `versionadded`)